docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,272 @@
+
+#### Connection Management
+
+Clients are expected to reconnect on error, when providing invalid `point`s for which a server cannot find an intersection.
+Clients may send multiple [`subscribe`](./messages/client/subscribe.md) messages for different accounts, but subsequently take responsibility to coordinate server-sent events by time to their possibly differently synchronized accounts. When receiving a message that references a `point` that the first account has already seen, the client only applies it for its second account for example.
+
+### Versioning
+
+In our protocol, the client sends its list of supported versions as part of the initial [`subscribe`](./messages/client/subscribe.md) message. Upon receiving this information, the server selects the highest mutually supported version and associates it with the client.
+This method requires the server to implement version-based message construction, allowing it to tailor its responses to each client's supported version.
+
+### Encoding
+
+Our proposed communication protocol **adopts blockchain-native encoding standards** to ensure compatibility and efficiency. This approach leverages existing conventions familiar to clients while minimizing conversion overhead on the server-side.
+
+#### Transaction Hash Encoding
+
+Transaction hashes are typically encoded using **hexadecimal representation**.
+
+#### Address Encoding
+
+Address encoding varies depending on the blockchain network. In Cardano the protocol servers bech32 encoded addresses.
+
+### Extensions
+
+Clients can turn on certain `extensions` by setting a respective flags in their subscription [`config](./messages/client/subscribe.md#subscribe-example) which tells the server to for example:
+
+- to enrich transaction output asset values by their respective off-chain metadata such as decimal offsets
+- to resolve transaction inputs (transaction output references) as transaction outputs
+
+## Rationale: how does this CIP achieve its goals?
+
+A primary consideration for the protocol design was to minimize message round trips between client and server as well as the amount of work to be done by the server for each client. Thus, the API's authentication does not require an initial message exchange but rather defines a standard verifable method for the server to either drop a client early and proceed to publish its relevant data.
+
+### Client Typed Messages
+
+- [`heartbeat`](./messages/client/heartbeat.md): Connection management/ keep alive
+- [`subscribe`](./messages/client/subscribe.md): Define topics of interest & authentication
+- [`unsubscribe`](./messages/client/unsubscribe.md): Graceful disconnect from server
```

Comment:
@iccicci If you have ten accounts in your wallet and want to remove one of them, you might want to be able to unsubscribe.

Reply:
Yes @ginnun , in theory yes... but which is the use case?

Since others can still send transactions to addresses bound to that account, what does it means "I want to remove an account"?

To me sounds as "if somebody still send me coins on one of these addresses, I'm no longer interested: it's fine if I do not track them".

When I want to stop using an account, I'll stop sharing addresses bound to it with other to ask for payments, but I'm still interested (much interested) in transactions which give me something.

What am I missing?
Thank you

Reply:
You are still the account owner, but you are no longer interested in monitoring it on a given subscription. One person can have multiple installations of the wallet, e.g., mobile and desktop, simultaneously. 

A wallet will probably allow you to remove a wallet address via the UI (Lace does), why should server still send events related to it? 

- After some time, you may want to decrease the number of subscriptions in your mobile wallet while keeping everything active on the desktop. You may want to hide it for some reason.
- You may want to drop unused accounts because you want to.

Reply:
Makes sense...
Probably we can just mark it as optional for the clients.